src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapGroupBySettings.jsx

```diff
@@ -29,26 +32,160 @@ const HeatmapGroupBySettings = () => {
+      })
+    );
+  }, [cellSetsOrder]);
+
+  const moveUp = (source, id) => {
+    const index = source.findIndex((e) => e.key === id);
+
+    const arr = [...source];
+
+    if (index <= 0) {
+      return arr;
+    }
+
+    const el = arr[index];
+    arr[index] = arr[index - 1];
+    arr[index - 1] = el;
+
+    return arr;
+  };
+
+  const moveDown = (source, id) => {
+    const index = source.findIndex((e) => e.key === id);
+
+    const arr = [...source];
+
+    if (index === -1 || index >= source.length - 1) {
+      return arr;
+    }
+
+    const el = arr[index];
+    arr[index] = arr[index + 1];
+    arr[index + 1] = el;
+
+    return arr;
+  };
+
+  const indexOfCellSet = (cellSet) => {
+    return cellSetsOrder.findIndex((elem) => (elem.key === cellSet.key));
+  }
+
+  const menu = (
+    <Menu>
+      {
+        getCellSets(['cellSets', 'metadataCategorical'])
+          .map((cellSet) => {
+            let positionInCellSetOrder = indexOfCellSet(cellSet);
+
+            return (
+              <Menu.Item size='small'>
+                <div style={{ margin: '0px', width: '100%' }} onClick={(e) => { e.stopPropagation(); }}>
+                  <Button
+                    shape="square"
+                    size='small'
+                    style={{ marginRight: '5px' }}
+                    icon={positionInCellSetOrder > -1 ? <MinusOutlined /> : <PlusOutlined />}
+                    onClick={() => {
+                      let newCellSetsOrder = [...cellSetsOrder];
+                      if (positionInCellSetOrder > -1) {
+                        // If the cell is included in the cellSet, we have to remove it
+                        newCellSetsOrder.splice(positionInCellSetOrder, 1);
+                      } else {
+                        // If the cell is not included in the cellSet, we have to add it
+                        newCellSetsOrder.push(cellSet);
+                      }
+
+                      setCellSetsOrder(newCellSetsOrder);
+                    }}
+                  >
+                  </Button>
+                  {cellSet.name}
+                </div>
+              </Menu.Item>
+            );
+          })
+      }
+    </Menu>
+  );
+
   return (
     <div style={{ padding: '5px' }}>
-      <Radio.Group
-        value={groupedTrack}
-        onChange={({ target }) => dispatch(
-          updatePlotConfig('interactiveHeatmap', {
-            groupedTrack: target.value,
-          }),
-        )}
-      >
-        {getCellSets(
-          ['cellSets', 'metadataCategorical'],
-        ).map((cellSet) => (
-          <Radio style={radioStyle} value={cellSet.key}>
-            {cellSet.name}
-          </Radio>
-        ))}
-      </Radio.Group>
-    </div>
+      <Space direction='vertical'>
+        <Dropdown overlay={menu}>
+          <div style={{ padding: '7px', border: '1px solid rgb(238,238,238)' }}>
+            Select the parameters to group by
```

Comment:
There is a usability issue with the functionality of this: After I have selected the parameters to group by, the menu does not disappear and show me my selections until I hover over the "group by" option in the main heatmap options menu. If I go out of the `Select the parameters to group by` window, the whole thing closes and then I have to go and click from scratch.

Reply:
I actually talked with vicky about how to go about this and I think she would prefer to just have the trigger action be click instead.

The other option I had thought about to deal with this was having the dropdown to open on the side instead of over the already existing component.

Reply:
I think that click will work, provided that the menu for selecting parameters collapses back after click, so that the user can see easily what they have selected showing up/disappearing.

I was actually wondering why did we not go with exactly what was implemented for the "metadata tracks" menu with the switches (see screenshot). @vickymorrison what are the reasons for why there we have switches and here we don't?
<img width="227" alt="Screenshot 2021-01-26 at 09 58 40" src="https://user-images.githubusercontent.com/65414651/105823137-132b6200-5fbd-11eb-8a7f-1da8fb75820f.png">

Reply:
My original wireframe design was exactly as you are suggesting, Iva. But @marcellp convinced me that the usability of that design was flawed - the switches imply that something is shown/hidden (like for the metadata tracks) and we thought that the function of the switches would be less clear and intuitive to users. This new design, as Martin is implementing, has much better usability and will also be better for large datasets with many metadata tracks.

Reply:
I see, I guess this is something that we will verify with the users. Are we having tow different designs for this for `metadata tracks` and `group by`, because we want to test with users which one is better?

Reply:
Yes, this is something that we can test and observe directly in usability studies at the open beta release.

Reply:
about the original issue, in the end I am leaving both hover and click actions as triggers, that way it is easy to close but also easy to open, let me know if it still feels weird and any other changes should be done.

Reply:
Has the staging updated? It doesn't seem to click to close the dropdown menu.

Reply:
For some reason it is not updating when I do new commits, I just created a new staging environment and set it in the description of the PR, once it is up it should have the latest changes.

Reply:
is that one fixed?

Reply:
yes, in the end we left it with both hover and click as trigger and vicky told me on slack it was ok.